test/services/objection.service.spec.unit.ts

```diff
@@ -35,12 +40,35 @@ describe("objections API service unit tests", () => {
   });
 
   it("returns undefined when adding an attachment", () => {
+    mockAddAttachment.mockImplementationOnce(() => {
+      return;
+    });
+    const session = {
+      data: {
+      },
+      } as Session;
+    session.data[OBJECTIONS_SESSION_NAME] = {
+      [SESSION_COMPANY_PROFILE] : dummyCompanyProfile,
+      [SESSION_OBJECTION_ID] : NEW_OBJECTION_ID,
+    };
     const FILE_NAME = "test_file";
-    const attachmentResult = objectionsService.addAttachment(COMPANY_NUMBER,
-        ACCESS_TOKEN,
-        NEW_OBJECTION_ID,
+    const attachmentResult = objectionsService.addAttachment(session,
         new Buffer(""),
         FILE_NAME );
-    expect(attachmentResult).toBeUndefined();
+
+    expect(attachmentResult).toEqual(Promise.resolve());
```

Comment:
i'd also add expect mockAddAttachment.toBeCalled (times(1)?)

Reply:
maybe check the buffer and filename are passed to mockAddAttachment would be better.